toolchain/parse/testdata/generics/interface/require_name.carbon

```diff
@@ -0,0 +1,41 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
```

Comment:
We've shifted towards using split-file tests (`// --- name.carbon`) for testing of a single feature, these single-test files are increasingly rare. For an example, see https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/parse/testdata/packages/import/export.carbon

You could probably use this to put `require_name` and `require_self` tests in the same file together, along with tests of incorrect syntax.